### PR TITLE
[New Package] Readarr

### PR DESCRIPTION
**Pull request by EckPhi**
## Description

motivation: New e-book collection manager in *arr flavour, Readarr.

Closes: #4801 

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Update-Policy#tests-checks)
- [ ] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created

### Type of change

- [ ] Bug fix
- [x] New Package
- [ ] Package update
- [ ] Includes small framework changes
- [ ] This change requires a documentation update (e.g. Wiki)

### CODE REVIEW

File: cross/readarr/Makefile
Comment (publicarray) [inline cross/readarr/Makefile:6]: Rather than nightly (I assume every day results in a new build?) is there a more stable version?

Comment (publicarray) [reply cross/readarr/Makefile:6]: `https://readarr.servarr.com/v1/update/develop/updatefile?os=linux&runtime=netcore&arch=x64` ?

File: spk/readarr/Makefile
Comment (publicarray) [inline spk/readarr/Makefile:2]:
```suggestion
SPK_VERS = 0.1.0.1225
```

IMHO Using a shell date is a last resort if no version number is available.

**Review by publicarray — CHANGES_REQUESTED** [review]
Welcome and Nice work :+1:  but please change the version number.

File: cross/readarr/Makefile
Comment (EckPhi) [reply cross/readarr/Makefile:6]: According to their [wiki](https://wiki.servarr.com/readarr/installation#linux): "Readarr is currently in beta testing and thus using the nightly branch."

**Comment by EckPhi** [comment]
Thank you! You are right I got the SPK_VERS from another package. 

Poses the fact that their release server only provides the latest release (at least I didn't find a way to request a specific version) a problem?

**Comment by publicarray** [comment]
@bakerboy448 @ta264 do you think the project is stable enough for production use on Synology devices? With the warning in the [wiki](https://wiki.servarr.com/readarr/installation#Linux) in mind:
>  Readarr is currently in beta testing and thus using the nightly branch. It is generally still in a work in progress. Features may be broken, incomplete, or cause spontaneous combustion.

I personally lean towards to wait until a release (more stable than nightly) release is done.

**Comment by bakerboy448** [comment]
Ageee on waiting for a develop

Coming soonish.

**Review by bakerboy448 — CHANGES_REQUESTED** [review]
Recommending waiting until a develop release drops and using that rather than nightly

File: spk/readarr/src/service-setup.sh
Comment (hgy59) [inline spk/readarr/src/service-setup.sh:6]: I suppose the comment should tell "config and home directories" as there is not PID-File definition below

File: spk/readarr/src/service-setup.sh
Comment (hgy59) [inline spk/readarr/src/service-setup.sh:33]: Please check the upgrade logs. I suppose that `service_postinst` is called when upgrading a package, so this part would be obsolete.

File: cross/readarr/Makefile
Comment (bakerboy448) [reply cross/readarr/Makefile:6]: Readarr has been formally develop as of early last week

**Comment by hgy59** [comment]
@publicarray as this is another "*arr" package that uses the internal updater I prefer to use date as SPK_VER.

File: spk/readarr/src/service-setup.sh
Comment (ta264) [inline spk/readarr/src/service-setup.sh:7]: IMO rather than guessing where the config is going to end up, you should pass the `-data=` argument to Readarr to force it to put it in the location you want

Comment (hgy59) [reply spk/readarr/src/service-setup.sh:7]: @ta264 is this for all *arr packages? _(I have to check the command line parameters...)_

Comment (bakerboy448) [reply spk/readarr/src/service-setup.sh:7]: Yes the \*arrs all support `-nobrowser -data=`

Comment (hgy59) [reply spk/readarr/src/service-setup.sh:7]: regarding the [radar wiki](https://wiki.servarr.com/radarr/appdata-directory#Argument) this is only required when running multiple instances of the same service.

Comment (bakerboy448) [reply spk/readarr/src/service-setup.sh:7]: > regarding the [radar wiki](https://wiki.servarr.com/radarr/appdata-directory#Argument) this is only required when running multiple instances of the same service.

Yes but that doesn't restrict it from being used for a single instance. And as Ta said, rather than guessing at things, just set it.

Comment (hgy59) [reply spk/readarr/src/service-setup.sh:7]: just wondered why none of the *arr package here does use this (not even radarr udpated by ta264 in #4464) 😄 
So I will update all *arr packages next.

Comment (ta264) [reply spk/readarr/src/service-setup.sh:7]: I was younger and less wise :D

The radarr improvements I'm working on do include that argument

Comment (hgy59) [reply spk/readarr/src/service-setup.sh:7]: We have a lot of copy&paste in the service-setup scripts of the *arr packages that need a brush up - so I will not only prevent the package downgrade but also add the -data parameter to the service command.

Comment (hgy59) [reply spk/readarr/src/service-setup.sh:7]: And redarding the *arr tools I am still a noob

Comment (mreid-tt) [reply spk/readarr/src/service-setup.sh:7]: > We have a lot of copy&paste in the service-setup scripts of the *arr packages that need a brush up - so I will not only prevent the package downgrade but also add the -data parameter to the service command.

I'm happy to take on that responsibility. Is the change that needs to be effected across `*arr` (Sonarr, Radarr, Lidarr, Prowlarr) packages the addition of the string "`-nobrowser -data=${CONFIG_DIR}`" at the end of the `SERVICE_COMMAND` parameter?

**Comment by mreid-tt** [comment]
@hgy59, @publicarray... I've gone through the code and updated it with the latest enhancements from the other `*arr` packages over the past few months and should be ready to merge. If the team can give it a last once over that would be great.

File: spk/readarr/Makefile
Comment (hgy59) [reply spk/readarr/Makefile:2]: This should be reverted to `$(shell date +%Y%m%d)` as all packages that use internal package updates should not define a specific version number.

Comment (mreid-tt) [reply spk/readarr/Makefile:2]: This change suggestion was not committed so nothing to revert at the moment. Should some other best practice be followed rather than resolving the conversation?

File: spk/readarr/Makefile
Comment (hgy59) [inline spk/readarr/Makefile:21]: the supporturl can be removed, as it is for BETA packages only

File: spk/readarr/Makefile
Comment (hgy59) [inline spk/readarr/Makefile:13]: Maintainer URL should match the Maintainer.

**Comment by mreid-tt** [comment]
@hgy59, @publicarray... while we await the full review of this package, there was a user who tested it and got an error with the internal auto-updater following installation. Their platform details are as follows:

```
DSM: DSM 7.1.1-42962 Update 1
Model name: DS216play
CPU: STM Monaco STiH412
```

Per their note (https://github.com/SynoCommunity/spksrc/issues/5574#issuecomment-1417759676), they received the following error in the Readarr log file:

```
2023-02-05 13:49:27.5|Info|InstallUpdateService|Starting update client /tmp/readarr_update/Readarr.Update
2023-02-05 13:49:27.5|Info|InstallUpdateService|Readarr will restart shortly.
2023-02-05 13:49:27.9|Info|InstallUpdateService|Updater Arguments: 29540 /tmp/readarr_update /volume1/@appstore/Readarr/Readarr
2023-02-05 13:49:29.3|Error|CommandExecutor|Error occurred while executing task ApplicationUpdate

[v0.1.2.1532] System.ComponentModel.Win32Exception (13): An error occurred trying to start process '/tmp/readarr_update/Readarr.Update' with working directory '/usr/syno/synoman/webapi'. Permission denied
```

Now I tried to replicate the error but could not. My platform details are a bit different:

```
DSM: DSM 7.1.1-42962 Update 2
Model name: VirtualDSM (DS916+)
CPU: INTEL Pentium N3710
```

My Readarr log file looks like this:

```
2023-02-11 09:57:56.6|Info|InstallUpdateService|Starting update client /volume1/@apptemp/readarr/readarr_update/Readarr.Update
2023-02-11 09:57:56.6|Info|InstallUpdateService|Readarr will restart shortly.
2023-02-11 09:57:56.6|Info|InstallUpdateService|Updater Arguments: 14991 /volume1/@apptemp/readarr/readarr_update /volume1/@appstore/readarr/share/Readarr/bin/Readarr /data=/volume1/@appdata/readarr/.config/Readarr /nobrowser
2023-02-11 09:58:17.7|Info|Bootstrap|Starting Readarr - /volume1/@appstore/readarr/share/Readarr/bin/Readarr.dll - Version 0.1.2.1558
```

What I notice is that the user's updater is starting in `/tmp/readarr_update` whereas my updater is starting in `/volume1/@apptemp/readarr/readarr_update`. Would this be contributing to the failed updater? If so, given there should be no significant changes in the build between his armv7-7.1 package and my x64-7.1 package, can you think of any reason for this?

**Comment by bakerboy448** [comment]
/tmp is not executable on DSM7

None of the updated packages would point to /tmp; sounds like they weren't running the package/version  they think they are?

**Comment by mreid-tt** [comment]
So I was testing this and while it installed find on DSM 6 and DSM 7 and did perform an auto-update, I feel that the way the package launches is different than the other `*arr` apps. On first install I see this line in the install logs:

```
2023/04/08 13:30:59	install readarr 20230408-1 Begin start-stop-status start
/var/packages/readarr/scripts/start-stop-status: line 151: 11599 Killed                  ${service} >> ${OUT} 2>&1
2023/04/08 13:32:19	install readarr 20230408-1 End start-stop-status start ret=[0]
```

Around the same time, the service log shows this:

```
[Debug] Readarr.Update: Starting /volume1/@apptemp/readarr/readarr_update/Readarr.Update 11599 /volume1/@apptemp/readarr/readarr_update /volume1/@appstore/readarr/share/Readarr/bin/Readarr /data=/volume1/@appdata/readarr/.config/Readarr /nobrowser 
[Debug] InstallUpdateService: Install in progress, giving installer 30 seconds. 
```

In the app once installed, it seems to be updated correctly. Don't know if this is a concern before merging. Don't see similar logs with Sonarr v4 on fresh install even though it's also a beta that auto-updates.

**Comment by mreid-tt** [comment]
I've pushed an update to 0.1.4.1596 because I saw in [0.1.2.1558](https://github.com/Readarr/Readarr/tree/v0.1.2.1558) there was a note on "_Don't block task queue for queued update task when long running tasks queued_". Re-running the clean install on DSM 6 and DSM 7 now has a clean install log.

@hgy59, @publicarray I believe this should be good to merge. Any objections to this?

**Review by mreid-tt — APPROVED** [review]

**Comment by hgy59** [comment]
@mreid-tt just updated the wiki page documenting the ports used by SynoCommunity packages (https://github.com/SynoCommunity/spksrc/wiki/SynoCommunity-Used-Ports)

**Pull request merged**
